### PR TITLE
CI: change go version to 1.20.4

### PR DESCRIPTION
**Pull request by guguducken**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [x] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # https://github.com/matrixorigin/MO-Cloud/issues/729

## What this PR does / why we need it:
update the version of go in ci to 1.20.4

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/matrixorigin/matrixone
 
-go 1.19
+go 1.20
```
Comment (fengttt) [inline go.mod:3]: you cannot do this because go mod tidy is not working.   Just keep this as 1.19

File: .github/actions/setup_env/action.yaml [hunk 2]
```diff
@@ -4,14 +4,14 @@ inputs:
   go-version:
     description: 'version of golang'
     required: false
-    default: "1.19"
+    default: "1.20.4"
```
Comment (fengttt) [inline .github/actions/setup_env/action.yaml:7]: do we want 1.20.4 or simply 1.20?

**Review by fengttt — COMMENTED** [review]
Two minor things.

File: go.mod (hunk 1 above)
Comment (guguducken) [reply go.mod:3]: Ok, I'll change it back

**Review by fengttt — APPROVED** [review]

**Review by sukki37 — APPROVED** [review]

**Pull request merged**